all_bin_scenario.py

```diff
@@ -236,13 +223,9 @@ def max_stepsize(t: float) -> float:
             value_off=None,
         )
 
-    ############# RUN FW BIN SIMUS #############
-    # TODO: adjust to run monoblocks in parallel
-    for nb_bin in range(3):
-        fw_bin = FW_bins.get_bin(nb_bin)
-        for sub_bin in fw_bin.sub_bins:
-            if sub_bin.material == "W":
-                my_model, quantities = make_W_mb_model(
+    def which_model(nb_bin: int, material: str):
```

Comment:
You still have a lot of code duplication here...

We'll look at this when the time for the big refactor comes but Seems to me we could do things differently here:

```python
material_to_model_function = {"W": make_W_mb_model, "B": make_B_mb_model, "SS": make_DFW_mb_model}

model_function = material_to_model_function[my_bin.material]
my_model, quantities = model_function(temperature=..., L=..., folder=...., ...)
```

Reply:
I guess the `temperature` argument also changes so need to have something slightly different but overall we need to fix this code duplication isseu